File: beginner/python/1478.py
while True:
    n = int(input())

    if n == 0:
        break

    iniMatriz = 0
    fimMatriz = n
    value = 1

    m = [[0] * n for _ in range(n)]

    for i in range(0,n):
        for j in range(0,n): 
            if i == j:
                m[i][j] = 1
            elif i < j:
                m[i][j] = j - i + 1
            else:
                m[i][j] = i - j + 1
    

    for i in range(n):
        rowStr = ''
        for j in range(n):
            if j == 0:
                rowStr += str(m[i][j]).rjust(3)
            else:
                rowStr += ' ' + str(m[i][j]).rjust(3)
        print(rowStr)

    print()
1

File: beginner/javascript/1478.js
let input = require('fs').readFileSync('stdin', 'utf-8').split('\n'); 

var i, j, n = 1, value, iniMatriz, fimMatriz;

while(n != 0){
    n = parseInt(input.shift());

    if(n == 0)
        break;
    
    iniMatriz = 0;
    fimMatriz = n;
    value = 1;

    var m = new Array();

    for(i = 0; i < n; i++) {
        m.push(new Array());
        for(j = 0; j < n; j++) {
            if (i == j)
                m[i][j] = 1;
            else if (i < j)
                m[i][j] = j - i + 1;
            else
                m[i][j] = i - j + 1;
        }
    }

    for (let i = 0; i < n; i++) {
        let rowStr = '';
        for (let j = 0; j < n; j++) {
            if (j == 0)
                rowStr += m[i][j].toString().padStart(3);
            else
                rowStr += ' ' + m[i][j].toString().padStart(3);
        }
        console.log(rowStr);
    }

    console.log();
}
